enos/modules/vault_verify_performance_replication/main.tf

```diff
@@ -0,0 +1,117 @@
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "vault_cluster_addr_port" {
+  description = "The Raft cluster address port"
+  type        = string
+  default     = "8201"
+}
+
+variable "vault_install_dir" {
+  type        = string
+  description = "The directory where the Vault binary will be installed"
+}
+
+variable "primary_leader_public_ip" {
+  type        = string
+  description = "Vault primary cluster leader Public IP address"
+}
+
+variable "primary_leader_private_ip" {
+  type        = string
+  description = "Vault primary cluster leader Private IP address"
+}
+
+variable "secondary_leader_public_ip" {
+  type        = string
+  description = "Vault secondary cluster leader Public IP address"
+}
+
+variable "secondary_leader_private_ip" {
+  type        = string
+  description = "Vault secondary cluster leader Private IP address"
+}
+
+# variable "primary_vault_root_token" {
+#   type        = string
+#   description = "The vault root token of primary cluster"
+# }
+
+# variable "secondary_vault_root_token" {
+#   type        = string
+#   description = "The vault root token of secondary cluster"
+# }
+
+variable "wrapping_token" {
+  type        = string
+  description = "The wrapping token created on primary cluster"
+  default     = null
+}
+
+locals {
+  primary_replication_status   = jsondecode(enos_remote_exec.verify_replication_on_primary.stdout)
+  secondary_replication_status = jsondecode(enos_remote_exec.verify_replication_on_secondary.stdout)
+}
+
+resource "enos_remote_exec" "verify_replication_on_primary" {
+  environment = {
+    VAULT_ADDR        = "http://127.0.0.1:8200"
+    vault_install_dir = var.vault_install_dir
+  }
+
+  scripts = ["${path.module}/scripts/verify-performance-replication.sh"]
+
+  transport = {
+    ssh = {
+      host = var.primary_leader_public_ip
+    }
+  }
+}
+
+output "primary_replication_status" {
+  value = local.primary_replication_status
+
+  precondition {
```

Comment:
These were introduced in Terraform 1.2.0. Do we need to update documentation/config or add a minimum Terraform version requirement to this module?

Reply:
We already have it https://github.com/hashicorp/vault/blob/main/enos/enos-terraform.hcl#L19